packages/data-context/src/codegen/spec-options.ts

```diff
@@ -18,18 +21,105 @@ interface CodeGenOptions {
 //   Button.foo.js  -> Button.foo-copy-1.js
 export const expectedSpecExtensions = ['.cy', '.spec', '.test', '-spec', '-test', '_spec']
 
+type ComponentExtension = '.cy.js' | '.cy.ts' | '.cy.jsx' | '.cy.tsx'
+type TemplateKey = 'e2e' | 'componentEmpty' | 'vue2Component' | 'vue3Component'
 export class SpecOptions {
   private parsedPath: ParsedPath;
+  private parsedErroredCodegenCandidate?: ParsedPath
+
+  private async getFrontendFramework () {
+    const detected = await detectFramework(this.ctx.currentProject || '')
+
+    return await detected.framework
+  }
 
   constructor (private ctx: DataContext, private options: CodeGenOptions) {
     assert(this.ctx.currentProject)
     this.parsedPath = this.ctx.path.parse(options.codeGenPath)
+
+    if (options.erroredCodegenCandidate) {
+      this.parsedErroredCodegenCandidate = this.ctx.path.parse(options.erroredCodegenCandidate)
+    }
   }
 
   async getCodeGenOptions () {
+    if (this.options.codeGenType === 'component') {
+      return this.getComponentCodeGenOptions()
+    }
+
+    return {
+      codeGenType: this.options.codeGenType,
+      fileName: await this.buildFileName(),
+      templateKey: this.options.codeGenType as TemplateKey,
+    }
+  }
+
+  private async getComponentCodeGenOptions () {
+    const frontendFramework = await this.getFrontendFramework()
+
+    if (!frontendFramework) {
+      throw new Error('Cannot generate a spec without a framework')
+    }
+
+    // This only works for Vue projects with default spec patterns right now. If the framework is not Vue, we're generating an empty component test
+    if (frontendFramework.codeGenFramework !== 'vue' || this.options.hasCustomSpecPattern) {
+      return {
+        codeGenType: this.options.codeGenType,
+        fileName: await this.buildFileName(),
+        templateKey: 'componentEmpty' as TemplateKey,
+      }
+    }
+
+    const frameworkOptions = await this.getFrameworkComponentOptions(frontendFramework)
+
+    return frameworkOptions
+  }
+
+  private relativePath () {
+    if (!this.parsedErroredCodegenCandidate?.base) {
+      return `./${this.parsedPath.base}`
+    }
+
+    const componentPathRelative = this.ctx.path.relative(this.parsedPath.dir, this.parsedErroredCodegenCandidate.dir)
+
+    const componentPath = this.ctx.path.join(componentPathRelative, this.parsedErroredCodegenCandidate.base)
+
+    return componentPath.startsWith('.') ? componentPath : `./${componentPath}`
+  }
+
+  private async getFrameworkComponentOptions (framework: WizardFrontendFramework) {
+    const componentName = this.parsedErroredCodegenCandidate?.name ?? this.parsedPath.name
+
+    const componentPath = this.relativePath()
+
+    const templateKey: TemplateKey = framework.mountModule === 'cypress/vue' ? 'vue3Component' : 'vue2Component'
+
     return {
       codeGenType: this.options.codeGenType,
-      fileName: await this.getFilename(),
+      componentName,
+      componentPath,
+      mountModule: framework.mountModule,
```

Comment:
Don't need to return `mountModule` since the template is handling the structure of the generated file.